File: QNet/src/main/java/com/qjuzi/qnet/pages/home/HomeActivity.java
package com.qjuzi.qnet.pages.home;

import com.qjuzi.architecure.tree.TreeActivity;
import com.qjuzi.architecure.tree.event.OnViewCreated;
import com.qjuzi.qnet.ActivityMainBinding;
import com.qjuzi.qnet.pages.home.model.HomeModel;
import com.qjuzi.qnet.pages.home.presenter.DelayTaskPresenter;
import com.qjuzi.qnet.pages.home.presenter.HomeMainPresenter;
import com.qjuzi.qnet.pages.home.presenter.TestLifecyclePresenter;

import android.os.Bundle;
import android.support.v7.widget.GridLayoutManager;

public class HomeActivity extends TreeActivity {
    public ActivityMainBinding binding;
    public HomeModel model;

    public HomeMainPresenter mainPresenter;
    public DelayTaskPresenter delayTaskPresenter;
    public TestLifecyclePresenter testLifecyclePresenter;

    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(binding.getRoot());
    }

    @OnViewCreated
    public void onViewCreated() {
        binding.recyclerView.setLayoutManager(new GridLayoutManager(this, 4));
    }

}

File: QNet/src/main/java/com/qjuzi/qnet/pages/home/HomeActivity.kt
package com.qjuzi.qnet.pages.home

import android.os.Bundle
import android.support.v7.widget.GridLayoutManager
import com.qjuzi.architecure.mvvm.MVVMActivity
import com.qjuzi.architecure.mvvm.OnViewCreated
import com.qjuzi.qnet.databinding.ActivityMainBinding
import com.qjuzi.qnet.pages.home.model.HomeModel
import com.qjuzi.qnet.pages.home.presenter.DelayTaskPresenter
import com.qjuzi.qnet.pages.home.presenter.HomeMainPresenter
import com.qjuzi.qnet.pages.home.presenter.TestLifecyclePresenter

class HomeActivity : MVVMActivity() {

    lateinit var binding: ActivityMainBinding
    lateinit var model: HomeModel


    lateinit var mainPresenter: HomeMainPresenter
    lateinit var delayTaskPresenter: DelayTaskPresenter
    lateinit var testLifecyclePresenter: TestLifecyclePresenter


    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(binding.root)
    }

    @OnViewCreated
    fun onViewCreated() {
        binding.recyclerView.layoutManager = GridLayoutManager(applicationContext, 4)
    }
}
